jboss/container/java/jvm/bash/artifacts/opt/jboss/container/java/jvm/container-limits

```diff
@@ -41,14 +74,27 @@ max_unbounded() {
 }
 
 container_memory() {
-  # High number which is the max limit unit which memory is supposed to be
-  # unbounded. 
-  local mem_file="/sys/fs/cgroup/memory/memory.limit_in_bytes"
   local max_mem_unbounded="$(max_unbounded)"
-  if [ -r "${mem_file}" ]; then
-    local max_mem="$(cat ${mem_file})"
-    if [ ${max_mem} -lt ${max_mem_unbounded} ]; then
-      echo "${max_mem}"
+  # High number which is the max limit unit which memory is supposed to be unbounded.
+  if [ "${CGROUPS_VERSION}" = "v1" ]; then
+    local mem_file="/sys/fs/cgroup/memory/memory.limit_in_bytes"
+    if [ -r "${mem_file}" ]; then
+      local max_mem="$(cat ${mem_file})"
+      if [ ${max_mem} -lt ${max_mem_unbounded} ]; then
+        echo "${max_mem}"
+      fi
+    fi
+  else
+    # v2
+    local mem_file="/sys/fs/cgroup/memory.max"
```

Comment:
This path is only correct if the processes' cgroup is the root one. I.e., the result of `awk -F: '{ print $3 }' /proc/$$/cgroup`.

Reply:
It might be that this is always true in the contexts you care about, i.e., container with a java process as PID 1. It's not true on my desktop, for example, when testing the script logic:
```
$ cat /sys/fs/cgroup/memory.max
cat: /sys/fs/cgroup/memory.max: No such file or directory
$ awk -F: '{ print $3 }' /proc/$$/cgroup
/user.slice/user-1000.slice/session-300.scope
$ cat /sys/fs/cgroup/$(awk -F: '{ print $3 }' /proc/$$/cgroup)/memory.max
max
```

Reply:
In containers, yes, I didn't spot any other different path, this should be enough.